concourse/enumerator.py

```diff
@@ -420,6 +420,9 @@ def __init__(
     def template_name(self):
         return self.pipeline_definition.get('template', 'default')
 
+    def _pipeline_name(self):
+        return '-'.join((self.pipeline_name, self.main_repo['branch']))
```

Comment:
- why make this method "private"?
- did you replace all usages of the `name` attr with this method (if not, please do)?

Reply:
Because its a sensible default and to prevent clashing with the `pipeline_name` attr. Thinking about this some more, the two names should be separated more clearly. One is given by the user (e.g. `cc-utils`) and referred to more often than the other which is rendered by us (`cc-utils-master`). The latter seems to be needed mostly in two places during replication.
I'll need to check further, but for all cases I could find we use the name as retrieved from Concourse.

Reply:
what do you mean by "clashing"? in python, attribute names must be unique. i.e. you cannot "overload" them (have a method named the same as a "plain" attr); I assume you knew that, and this is what you meant by "avoidance"

so if I remember properly, the plain attr `pipeline_name` is the value from the `pipeline_definition` file, whereas the (new) method `_pipeline_name()` calculates the derived pipeline-names actually to be used for the rendered pipelines.

in that case, I would rather name the method to express this semantics. such as `unique_pipeline_name`, or `effective_pipeline_name`, and add a - brief - comment as to why we add e.g. the branch name.

for consistency reasons, we might also consider to do the name-calculation before the object is created; not sure whether the benefit is worth the efforts, tnough